### PR TITLE
add additional param to flag<fund> to set compared date

### PR DESCRIPTION
**Pull request by TactfulYuu**
### Enhancement of flag **fund**
- add new usage to flag **fund**: e.g. /fund fund_id compared_date
    - new key in bot response `净值对比{compared_data}增长`
- new data source which can get fund data in specific time range: https://fundf10.eastmoney.com/F10DataApi.aspx?type=lsjz&code={fund_id}&page=1&per=1&sdate={start_date}&edate={end_date}
- new decode tools: BeautifulSoup

### Reconstruct the proj
- move redundant func in data_source.py to utils in fund
- move handler impl to models
- process data in models rather in data_source

### CODE REVIEW

File: fundbot/util.py [hunk 1]
```diff
@@ -3,3 +3,11 @@
 
 def get_random_dt():
     return int(time.time()*1000 - random.randint(1, 500))
+
+def get_timestamp_from_dt(date):
+    try:
+        time_array = time.strptime(date, "%Y-%m-%d")
+        time_stamp = int(time.mktime(time_array)*1000 - random.randint(1, 500))
+    except Exception as e:
```
Comment (laazy) [inline fundbot/util.py:11]: meanless `try ... except` ?

File: fundbot/plugins/fund/fund.py [hunk 2]
```diff
@@ -15,10 +15,15 @@ async def handle_fund(bot: Bot, event: Event, state: T_State):
     args = str(event.get_message()).split()
     result = ""
     if len(args) == 0:
-        await fund.finish("用法: #fund <fundid>")
```
Comment (laazy) [inline fundbot/plugins/fund/fund.py:18]: I suggest that separate these two usage, and make them two command, such as `fund` and `fund_hist`.

File: fundbot/plugins/fund/data_source.py [hunk 3]
```diff
@@ -51,6 +58,37 @@ async def get_fund_data(fund_id: str) -> str:
             return "卧槽查不到啊"
 
 
+async def get_fund_trend_data(fund_id: str, start_date: str) ->  str:
+    recent_time = util.get_random_dt()
+    try:
```
Comment (laazy) [inline fundbot/plugins/fund/data_source.py:63]: process exception in util will be better.

File: fundbot/plugins/fund/data_source.py [hunk 4]
```diff
@@ -2,9 +2,11 @@
 from functools import wraps
 from typing import Union, List, Dict
 import traceback
+import asyncio
 
 import httpx
 from nonebot.log import logger, default_format
+from bs4 import BeautifulSoup
```
Comment (laazy) [inline fundbot/plugins/fund/data_source.py:9]: should modify `requirements.txt` when import new package.

**Review by laazy — CHANGES_REQUESTED** [review]

File: fundbot/util.py (hunk 1 above)
Comment (TactfulYuu) [reply fundbot/util.py:11]: Exception will be caught in func get_fund_trend_data

File: fundbot/plugins/fund/data_source.py (hunk 4 above)
Comment (TactfulYuu) [reply fundbot/plugins/fund/data_source.py:9]: plz refer https://github.com/CommonProsperity/fundbot/blob/feature/time_range/requirements.txt#L4

File: fundbot/util.py (hunk 1 above)
Comment (laazy) [reply fundbot/util.py:11]: if so, you don't need to write `try ... except`.

File: fundbot/plugins/fund/data_source.py (hunk 4 above)
Comment (laazy) [reply fundbot/plugins/fund/data_source.py:9]: sorry, I didn't aware that.

**Review by laazy — APPROVED** [review]

**Pull request merged**